kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ChecksumStatsSuite.scala

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import java.util.{Collections, Optional}
+
+import scala.jdk.CollectionConverters.{asJavaIteratorConverter, mapAsJavaMapConverter, seqAsJavaListConverter, setAsJavaSetConverter}
+
+import io.delta.kernel.{Table, Transaction, TransactionCommitResult}
+import io.delta.kernel.data.Row
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.internal.{InternalScanFileUtils, SnapshotImpl, TableConfig, TableImpl}
+import io.delta.kernel.internal.actions.{AddFile, GenerateIcebergCompatActionUtils, RemoveFile}
+import io.delta.kernel.internal.checksum.ChecksumReader
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.stats.FileSizeHistogram
+import io.delta.kernel.internal.util.Utils.toCloseableIterator
+import io.delta.kernel.utils.{CloseableIterable, DataFileStatus}
+import io.delta.kernel.utils.CloseableIterable.inMemoryIterable
+
+/**
+ * Functional e2e test suite for verifying file stats collection in CRC are correct.
+ */
+class ChecksumStatsSuite extends DeltaTableWriteSuiteBase {
+
+  test("Check stats in checksum are correct") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      createEmptyTable(
+        engine,
+        tablePath,
+        testSchema,
+        // Currently only table with IcebergWriterCompatV1 could easily
+        // support both add/remove files.
+        tableProperties = Map(
+          TableConfig.ICEBERG_WRITER_COMPAT_V1_ENABLED.getKey -> "true"))
+      val expectedFileSizeHistogram = FileSizeHistogram.createDefaultHistogram()
+
+      addFiles(
+        engine,
+        tablePath,
+        Map("file1.parquet" -> 100, "file2.parquet" -> 100802),
+        expectedFileSizeHistogram)
+      checkCrcCorrect(
+        engine,
+        tablePath,
+        version = 1,
+        expectedFileCount = 2,
+        expectedTableSize = 100902,
+        expectedFileSizeHistogram = expectedFileSizeHistogram)
+
+      removeFiles(
+        engine,
+        tablePath,
+        Map("file1.parquet" -> 100),
+        expectedFileSizeHistogram)
+      checkCrcCorrect(
+        engine,
+        tablePath,
+        version = 2,
+        expectedFileCount = 1,
+        expectedTableSize = 100802,
```

Comment:
idea / question: would it be clearer to just write `100902 - 100`? I trust whatever you think 👍